testutils/pkg.go

```diff
@@ -27,6 +27,17 @@ type TestPackage struct {
 	build  *buildObj
 }
 
+// Option provides a way to adjust the package config depending on testing
+// requirements
+type Option func(conf *packages.Config)
+
+// WithBuildTags enables injecting build tags into the package config on build.
+func WithBuildTags(tags []string) Option {
```

Comment:
To not break the API for existing tests, I decided to use variadic options to inject expected build tags under test.